assignments/introductory-assignments.md

```diff
@@ -488,14 +582,767 @@ A digital circuit computer with 2GB of memory can distinguish 2^{17 179 869 184}
+
+We execute `selfie.m` by first loading it using the `-l` option and then running it by invoking mipster using the -m` option:
+
+```sh
+./selfie -l selfie.m -m 1
+```
+
+**Output**:
+
+```sh
+./selfie: 171968 bytes with 38720 instructions and 12992 bytes of data loaded from selfie.m
+./selfie: selfie executing selfie.m with 1MB physical memory on mipster
+synopsis: selfie.m { -c { source } | -o binary | [ -s | -S ] assembly | -l binary } [ ( -m | -d | -r | -y ) 0-4096 ... ]
+./selfie: selfie.m exiting with exit code 0 and 0.00MB mallocated memory
+./selfie: selfie terminating selfie.m with exit code 0
+./selfie: summary: 65383 executed instructions [20.45% nops] and 0.17MB(17.57%) mapped memory
+./selfie: init:    lui: 27(0.04%), addi: 27228(41.64%)
+./selfie: memory:  ld: 14518(22.20%), sd: 9197(14.06%)
+./selfie: compute: add: 1574(2.40%), sub: 1433(2.19%), mul: 1740(2.66%)
+./selfie: compute: divu: 610(0.93%), remu: 379(0.57%)
+./selfie: compare: sltu: 858(1.31%)
+./selfie: control: beq: 1121(1.71%), jal: 4460(6.82%), jalr: 2107(3.22%)
+./selfie: system:  ecall: 131(0.20%)
+./selfie: profile: total,max(ratio%)@addr,2max,3max
+./selfie: calls:   2107,611(28.99%)@0x34E0,327(15.51%)@0x36F4,327(15.51%)@0x3E0C
+./selfie: loops:   184,75(40.76%)@0x56E4,63(34.23%)@0x194,45(24.45%)@0x4F08
+./selfie: loads:   14518,611(4.20%)@0x34F4,611(4.20%)@0x34F8,611(4.20%)@0x3508
+./selfie: stores:  9197,611(6.64%)@0x34E4,611(6.64%)@0x34EC,327(3.55%)@0x36F8
+./selfie: CPU+memory:    reads+writes,reads,writes[reads/writes]
+./selfie: heap segment:  862,674,188[3.58]
+./selfie: gp register:   2810,2809,1[2809.00]
+./selfie: data segment:  3116,3093,23[134.47]
+./selfie: ra register:   8172,4086,4086[1.00]
+./selfie: sp register:   44852,29129,15723[1.85]
+./selfie: s0 register:   14231,10272,3959[2.59]
+./selfie: stack total:   67255,43487,23768[1.82]
+./selfie: stack segment: 19738,10751,8987[1.19]
+./selfie: a0 register:   6093,1996,4097[0.48]
+./selfie: a1 register:   121,0,121[0.00]
+./selfie: a2 register:   121,0,121[0.00]
+./selfie: a6 register:   1,0,1[0.00]
+./selfie: a7 register:   131,0,131[0.00]
+./selfie: args total:    6467,1996,4471[0.44]
+./selfie: t0 register:   28395,14202,14193[1.00]
+./selfie: t1 register:   12774,6394,6380[1.00]
+./selfie: t2 register:   3076,1538,1538[1.00]
+./selfie: t3 register:   252,126,126[1.00]
+./selfie: temps total:   44497,22260,22237[1.00]
+```
+
+After loading `selfie.m`, the `-m 1` option directs mipster to emulate a computer with 1 megabyte of memory (abbreviated 1MB, explained below) for executing `selfie.m`. Since `selfie.m` is invoked without any options, which could appear after the `-m 1 option, it responds, just like selfie without options before, with its usage pattern and then terminates after that mipster terminates and outputs a summary of its built-in performance profiler.
```

Comment:
```suggestion
After loading `selfie.m`, the `-m 1` option directs mipster to emulate a computer with 1 megabyte of memory (abbreviated 1MB) for executing `selfie.m`. Since `selfie.m` is invoked without any options, which could appear after the `-m 1` option, it responds, just like selfie without options before, with its usage pattern and then terminates after that mipster terminates and outputs a summary of its built-in performance profiler.
```